token-tracker/src/truth-sources/lp-tokens.js

```diff
@@ -0,0 +1,275 @@
+/** @typedef { import("../lib/context").Context } Context */
+/** @typedef { import("../lib/ethereum-helper").Address } Address */
+
+import BN from "bn.js"
+
+import { ITruthSource } from "./truth-source.js"
+import { getPastEvents, getChainID } from "../lib/ethereum-helper.js"
+import { dumpDataToFile } from "../lib/file-helper.js"
+import { logger } from "../lib/winston.js"
+import { getPairData } from "../lib/uniswap.js"
+import { EthereumHelpers } from "@keep-network/tbtc.js"
+
+// https://etherscan.io/address/0xe6f19dab7d43317344282f803f8e8d240708174a#code
+import keepEthTokenJson from "../../artifacts/KEEP-ETH-UNI-V2-Token.json"
+// https://etherscan.io/address/0x38c8ffee49f286f25d25bad919ff7552e5daf081#code
+import keepTbtcTokenJson from "../../artifacts/KEEP-TBTC-UNI-V2-Token.json"
+
+import LPRewardsKEEPETHJson from "@keep-network/keep-ecdsa/artifacts/LPRewardsKEEPETH.json"
+import LPRewardsKEEPTBTCJson from "@keep-network/keep-ecdsa/artifacts/LPRewardsKEEPTBTC.json"
+
+// https://info.uniswap.org/pair/0xe6f19dab7d43317344282f803f8e8d240708174a
+const KEEPETH_PAIR_ADDRESS = "0xe6f19dab7d43317344282f803f8e8d240708174a"
+// https://info.uniswap.org/pair/0x38c8ffee49f286f25d25bad919ff7552e5daf081
+const KEEPTBTC_PAIR_ADDRESS = "0x38c8ffee49f286f25d25bad919ff7552e5daf081"
+
+// https://etherscan.io/tx/0xc64ac175846e719bb4f7f9b17a0b04bc365db3dda9d97ef70d7ede8f9c1a265b
+const KEEPETH_CREATION_BLOCK = "10100034"
+// https://etherscan.io/tx/0x1592f9b235c602c87a5b8cc5f896164dc43d16b92664cb9c8b420d28b64ca4a0
+const KEEPTBTC_CREATION_BLOCK = "11452642"
+
+const KEEP_IN_LP_KEEPETH_BALANCES_PATH =
+  "./tmp/keep-in-lp-keepeth-token-balances.json"
+
+const KEEP_IN_LP_KEEPTBTC_BALANCES_PATH =
+  "./tmp/keep-in-lp-keeptbtc-token-balances.json"
+
+export class LPTokenTruthSource extends ITruthSource {
+  /**
+   * @param {Context} context
+   * @param {Number} targetBlock
+   */
+  constructor(context, targetBlock) {
+    super(context, targetBlock)
+  }
+
+  async initialize() {
+    const chainID = await getChainID(this.context.web3)
+
+    const keepEthTokenContract = EthereumHelpers.buildContract(
+      this.context.web3,
+      JSON.parse(keepEthTokenJson.result),
+      KEEPETH_PAIR_ADDRESS
+    )
+
+    const lpRewardKeepEthContract = EthereumHelpers.buildContract(
+      this.context.web3,
+      LPRewardsKEEPETHJson.abi,
+      LPRewardsKEEPETHJson.networks[chainID].address
+    )
+
+    const keepTbtcTokenContract = EthereumHelpers.buildContract(
+      this.context.web3,
+      JSON.parse(keepTbtcTokenJson.result),
+      KEEPTBTC_PAIR_ADDRESS
+    )
+
+    const lpRewardKeepTbtcContract = EthereumHelpers.buildContract(
+      this.context.web3,
+      LPRewardsKEEPTBTCJson.abi,
+      LPRewardsKEEPTBTCJson.networks[chainID].address
+    )
+
+    this.liquidityStakingObjects = {
```

Comment:
Nitpick, but maybe we should rename `liquidityStakingObjects`to `liquidityPoolsInfo` or `liquidityPools`?

These objects do not refer to keep LP staking contracts.

Reply:
At first I actually named it `liquidityPool` but within this object, we have a LPRewardContract which handles staking. When saying a liquidity pool, I think of a pool like Uniswap that holds LP tokens, but here we have a notion of "staking" those LP tokens. Anyway, I'm okay to name it just like you proposed, but just want to double check it.

Reply:
So maybe just `liquidityStaking` or `liquidityStakingInfo`? Not a big deal, just that `*Objects` reads weird 🙈

Reply:
ok e6c3644